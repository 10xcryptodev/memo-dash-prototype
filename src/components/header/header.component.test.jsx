import React from 'react'
import { shallow } from 'enzyme'
import HeaderComponent from './header.component'
import { Menu } from 'semantic-ui-react'

describe('<HeaderComponent />', () => {
  let wrapper

  beforeEach(() => {
    const div = document.createElement('div')
    document.body.appendChild(div)
    wrapper = shallow(<HeaderComponent location="/" onHomeClicked={jest.fn()} />)
  })

  describe('should render', () => {
    it('without crashing', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('interaction', () => {
    const findMenuItem = (wrapper, text) =>
      wrapper
        .find(Menu.Item)
        .findWhere(n => n.text() === text)
        .parent()

    let spies

    beforeEach(() => {
      spies = {
        onHomeClicked: jest.fn(),
        onUsersClicked: jest.fn()
      }

      wrapper = shallow(
        <HeaderComponent
          location="/"
          onHomeClicked={spies.onHomeClicked}
          onUsersClicked={spies.onUsersClicked}
        />
      )
    })

    it('should call onHomeClicked when clicked on home', () => {
      findMenuItem(wrapper, 'Home').simulate('click')
      expect(spies.onHomeClicked).toHaveBeenCalled()
    })

    it('should call onHomeClicked when clicked on logo', () => {
      findMenuItem(wrapper, 'MemoDash').simulate('click')
      expect(spies.onHomeClicked).toHaveBeenCalled()
    })

    it('should call onUsersClicked when clicked on users', () => {
      findMenuItem(wrapper, 'Users').simulate('click')
      expect(spies.onUsersClicked).toHaveBeenCalled()
    })
  })
})
