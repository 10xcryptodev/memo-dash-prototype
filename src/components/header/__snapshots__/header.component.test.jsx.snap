// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderComponent /> should render without crashing 1`] = `
<header>
  <Menu
    color="blue"
    fixed="top"
  >
    <Container>
      <MenuItem
        as="a"
        header={true}
        onClick={[MockFunction]}
      >
        <Logo />
        MemoDash
      </MenuItem>
      <MenuItem
        active={false}
        as="a"
        onClick={[MockFunction]}
      >
        Home
      </MenuItem>
      <MenuItem
        active={false}
        as="a"
      >
        Users
      </MenuItem>
      <MenuMenu
        position="right"
      >
        <MenuItem>
          <Connect(SearchComponent) />
        </MenuItem>
        <Connect(UserMenuComponent) />
      </MenuMenu>
    </Container>
  </Menu>
</header>
`;
